river/drift/retrain.py

```diff
@@ -99,3 +101,77 @@ def _unit_test_params(cls):
             "drift_detector": drift.binary.DDM(),
         }
         yield {"model": naive_bayes.GaussianNB(), "drift_detector": drift.binary.DDM()}
+
+
+class DriftRetrainingRegressor(base.Wrapper, base.Regressor):
+    """Drift retraining regressor.
+
+    This regressor is a wrapper for any regressor. It monitors the incoming data for concept
+    drifts in the model's predictions. In case a drift is detected, the model will be reset.
+
+    Parameters
+    ----------
+    model
+        The classifier and background classifier class.
+    drift_detector
+        Algorithm to track concept drifts.
+
+    Examples
+    --------
+
+    >>> from river import drift
+    >>> from river import datasets
+    >>> from river import evaluate
+    >>> from river import tree
+    >>> from river import metrics
+
+    >>> dataset = datasets.synth.FriedmanDrift(drift_type='gra', position=(500, 700), transition_window= 50, seed=42).take(3000)
+
+    >>> model = drift.DriftRetrainingRegressor(model = tree.HoeffdingTreeRegressor())
+
+    >>> metric = metrics.RMSE()
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    RMSE: 3.061178
+    """
+
+    def __init__(self, model: base.Regressor| None = None, drift_detector: base.DriftDetector | None = None):
+        self.model = model if model is not None else linear_model.BayesianLinearRegression()
+        self.drift_detector = drift_detector if drift_detector is not None else drift.ADWIN()
+        self._var = stats.Var()
+        self.n_drifts_detected = 0
+
+    @property
+    def _wrapped_model(self):
+        return self.model
+
+    def predict_one(self, x):
+        return self.model.predict_one(x)
+
+    def learn_one(self, x, y):
+        self._update_ddm(x, y)
+        self.model.learn_one(x, y)
+        return self
+
+    # updates drift detector just as in forest/AdaptiveRandomForest
+    def _update_ddm(self, x, y):
+        y_pred = self.model.predict_one(x)
+        if y_pred is None:
+            return
+
+        drift_input = y - y_pred
+        self._var.update(drift_input)
+        if self._var.mean.n == 1:
+            self.drift_detector.update(0.5)
+        else:
+            sd = sqrt(self._var.get())
+            # We assume the error follows a normal distribution -> (empirical rule)
+            # 99.73% of the values lie  between [mean - 3*sd, mean + 3*sd]. We
+            # assume this range for the normalized data. Hence, we can apply the
+            # min-max norm to cope with  ADWIN's requirements
+            self.drift_detector.update((drift_input + 3 * sd) / (6 * sd) if sd > 0 else 0.5)
```

Comment:
Should these constants be passed as parameters to the class?

Also, do you have any paper/reference to back up this heuristic? It seems reasonable, but I can imagine other ways of doing this. For instance, using quantiles.